pkg/generate/app/pipeline.go

```diff
@@ -311,6 +311,30 @@ func objectMetaData(raw interface{}) (runtime.Object, *kapi.ObjectMeta, error) {
 	return obj, meta, nil
 }
 
+type acceptBuildConfigs struct {
+	typer runtime.ObjectTyper
+}
+
+// Accept accepts just BuildConfigs
+func (a *acceptBuildConfigs) Accept(from interface{}) bool {
+	obj, _, err := objectMetaData(from)
+	if err != nil {
+		return false
+	}
+	_, kind, err := a.typer.ObjectVersionAndKind(obj)
+	if err != nil {
+		return false
+	}
+	return kind == "BuildConfig"
```

Comment:
What about the output image stream? or the input image stream if creating from a docker repo?